### PR TITLE
multi: check rpc semver compatibility

### LINKED ISSUE #1151: dexcctl: Add supported rpc version.
If dexc and dexcctl are built from a different commit, dexcctl may work and it may not. The only way to know is to try methods until maybe you hit an error. For this reason I propose adding the supported rpc version to -V returns so that a user can know if their dexcctl should be compatible with the rpcserver they are communicating with. This version should be kept up to date with the dexc rpcserver version.

More discussion here: https://github.com/decred/dcrdex/pull/1149#discussion_r682739382

### PR DESCRIPTION
**Pull request by ukane-philemon**
This adds new values to RPC server version return. Closes #1151 
- add dexc version to rpc server version return value
- add supported RPC version to dexcctl.
- change dexc RPC server version return to a json object
- Edit: update rpc version handler test
```
$ ./dexcctl version
{
  "rpcServerVersion": {
    "Major": 0,
    "Minor": 2,
    "Patch": 0
  },
  "dexcVersion": {
    "versionString": "0.5.0-pre+9763b71bb",
    "major": 0,
    "minor": 5,
    "patch": 0,
    "prerelease": "pre",
    "buildMetadata": "9763b71bb.go1-18"
  }
}
```

```
$ ./dexcctl -V
dexcctl version 0.5.0-pre+9763b71bb (Go version go1.18 linux/amd64)
dexcctl required RPC server version: 0.2.0
```

### CODE REVIEW

**Review by JoeGruffins — APPROVED** [review]
Looks perfect to me.

File: client/rpcserver/rpcserver.go [hunk 1]
```diff
@@ -31,16 +31,17 @@ import (
 )
 
 const (
+	// rpcSemver is the RPC server's semantic API version.
+	// Move major up one for breaking changes. Move minor for
+	// backwards compatible features. Move patch for bug fixes.
+	// Dexcctl requiredRPCSemVer should be kept up to date with this version.
+	rpcSemverMajor uint32 = 0
+	rpcSemverMinor uint32 = 2
+	rpcSemverPatch uint32 = 0
```
Comment (chappjc) [inline client/rpcserver/rpcserver.go:40]: Please move back and rewrap the comment

File: dex/version/version.go [hunk 2]
```diff
@@ -106,3 +107,17 @@ func Parse(version string) string {
 
 	return version
 }
+
+// NormalizeString returns the passed string stripped of all characters which
+// are not valid according to the semantic versioning guidelines for pre-release
+// and build metadata strings.  In particular they MUST only contain characters
+// in semanticAlphabet.
+func NormalizeString(str string) string {
+	var result bytes.Buffer
```
Comment (chappjc) [inline dex/version/version.go:116]: `bytes.Buffer` -> `strings.Builder`

**Review by chappjc — CHANGES_REQUESTED** [review]
Seems good.  A few minor things.

**Review by chappjc — APPROVED** [review]

**Pull request merged**